### ISSUE TITLE
Integrate wordnet english

### ISSUE THREAD

**Issue by LifeIsStrange** [nltk/nltk#2977]
[English wordnet](https://github.com/globalwordnet/english-wordnet) is a fork of princeton wordnet that is developped in opensource, as such it is much more actively maintained, hence it is both more correct and more complete.
NLTK integrating english wordnet corpus as an option, would make it more discoverable and accessible.

@tomaarsen friendly ping 

The complexity seems low, as if wordnet english can be exported to the same format as princeton wordnet then nltk would support wordnet english out of the box?
If so what format does NLTK expect?

**Comment by tomaarsen** [comment]
I believe that NLTK can use the english-wordnet data. For example see https://github.com/globalwordnet/english-wordnet/issues/771 and https://github.com/nltk/nltk/pull/2860 by @ekaf. 

Alongside that open-source english Wordnet, @goodmami is working on [wn](https://github.com/goodmami/wn), which may also be worth checking out as an alternative to NLTK's Wordnet functionality. As far as I can tell, there's some good work being done there. 

However, to put this all in a broader context, the NLTK team has been looking to deprecate the existing NLTK Wordnet corpus reader and opt for a separate one for quite some while now (e.g. #2423). That said, I don't believe there is consensus on how to move forward with this deprecation fully. Wordnet is one of the big use cases of NLTK, so we need to be careful here. 
In short, this is a difficult discussion, but I imagine we would eventually move to deprecate the NLTK Wordnet in favor of `wn`, if the latter can be maintained and improved more effectively. Or perhaps we would just leave the choice with our users. There is no consensus on this yet, I believe.

I hope that helps clarify everything somewhat.

**Comment by LifeIsStrange** [comment]
Thank you for this great answer :). While NLTK and wn supports english wordnet, english wordnet has 2 upstream issues (including one regarding sense ordering, as pointed by ekaf) that are not yet fixed so in the meantime I will stick with NLTK classic wordnet 3.0/3.1.

**Comment by jmccrae** [comment]
I was wondering if we can reopen this or update a new issue for this? Open English WordNet has had two stable version released since this and in the meanwhile WordNet 3.0 has not gotten any better.

**Comment by ekaf** [comment]
@jmccrae, would you agree that NLTK only needs one version of the OEWN, and that the latest version could deprecate the previous ones?
That would make it much easier to stay up to date, because only the db in nltk_data would need upgrading while the handler could stay the same in nltk, where the OEWN version identifier is extracted from the copyright notice.

**Comment by jmccrae** [comment]
Sure, we aim for the most recent version to be the best one so that is ideal. 

For scientific reproducibility, it may be good to give some documentation on loading historical versions, if users desire.

**Comment by ekaf** [comment]
Thanks @jmccrae, avoiding the proliferation of deprecated db versions was also discussed in #3059. 
At this moment, NLTK has a wordnet2022 package so it only lacks support for the 2023 Edition. 
Looking forward, we will need a generic package name for the latest OEWN version. It could be as simple as "oewn", or maybe it would better to include "wordnet" in the package name. Please don't hesitate to share any suggestions.

**Comment by jmccrae** [comment]
`english-wordnet` is the name of the GitHub repo, so we could use that?

**Comment by ekaf** [comment]
> `english-wordnet` is the name of the GitHub repo, so we could use that?

Sure, _english-wordnet_ is the ideal name.
